### PR TITLE
Fixes #213 - Update the modify skill page as per new API

### LINKED ISSUE #213: Update the modify skill page as per new API
From the modify skill page user must be able to change the name of the skill, the path of the skill, the image of the skill, the image name of the skill and the content of the skill.
The UI must be same as the CreateSkill Component.

### PR DESCRIPTION
**Pull request by rishiraj824**
Fixes issue #213

Changes:
- [x] Modified the UI and added functionality to edit Model, Group values
- [x] Edit form data which is being sent.
- [x] Preview the old image and the new image
- [x] Fetch old values when clicking on the skill

http://periodic-trousers.surge.sh

@dynamitechetan @uday96 @saurabhjn76 @madhavrathi @isuruAb  Please check once.

### CODE REVIEW

File: src/components/SkillEditor/SkillEditor.js [hunk 1]
```diff
@@ -148,82 +189,225 @@ export default class Container extends React.Component {
...
+                            </SelectField>
+                            <TextField
+                                floatingLabelText="Enter Skill name"
+                                floatingLabelFixed={true}
+                                style={{marginLeft:10,marginRight:10}}
+                                value={this.state.expertValue}
+                                onChange={this.handleExpertChange}
+                            />
+                            { this.state.showImage &&  <img alt="preview" id="target" style={{width:60,height:60,borderRadius:"50%",marginRight:20,border: 0}}src={this.state.image}/> }
+                            <RaisedButton
```
Comment (uday96) [inline src/components/SkillEditor/SkillEditor.js]: Make the `Choose Image` button color also match the top bar color

**Comment by rishiraj824** [comment]
@uday96 Changed the color. Please check.
@sudheesh001 Squashed the commits as well.

File: src/components/SkillEditor/SkillEditor.js (hunk 1 above)
Comment (rishiraj824) [inline src/components/SkillEditor/SkillEditor.js]: @uday96 Changed the button color! Thanks :+1:

File: src/components/SkillEditor/SkillEditor.js [hunk 2]
```diff
@@ -52,16 +53,45 @@ export default class Container extends React.Component {
...
+                codeChanged:true
+            });
+        }
         self.updateCode(newValue)
     }
 
     constructor(props) {
-        fromListing = false;
         super(props);
+        console.log(this.props.location.state);
```
Comment (sudheesh001) [inline src/components/SkillEditor/SkillEditor.js]: You may want to remove the console statement here

File: src/components/SkillEditor/SkillEditor.js [hunk 3]
```diff
@@ -52,16 +53,45 @@ export default class Container extends React.Component {
...
+            expertValue:this.props.location.state.oldExpertValue,
+            oldExpertValue: this.props.location.state.oldExpertValue,
+            oldImageUrl:this.props.location.state.oldImageValue,
+            imageUrl:this.props.location.state.oldImageValue,
+            image_name_changed:false,
+            code:"::name <Skill_name>\n::author <author_name>\n::author_url <author_url>\n::description <description> \n::dynamic_content <Yes/No>\n::developer_privacy_policy <link>\n::image <image_url>\n::terms_of_use <link>\n\n\nUser query1|query2|quer3....\n!example:<The question that should be shown in public skill displays>\n!expect:<The answer expected for the above example>\nAnswer for the user query",
+            fontSizeCode:14,
+            editorTheme:"github"
         };
+        console.log(this.state);
```
Comment (sudheesh001) [inline src/components/SkillEditor/SkillEditor.js]: Same here.

File: src/components/SkillEditor/SkillEditor.js [hunk 4]
```diff
@@ -148,82 +189,227 @@ export default class Container extends React.Component {
...
+                loading:false
+            });
+            return 0;
+        }
+        let file;
 
-        let url= "http://api.susi.ai/cms/modifySkill.json?"+skill_relative_path+"&content="+encodeURIComponent(this.state.code)+"&changelog="+this.state.commitMessage+"&access_token="+cookies.get('loggedIn');
+        var form = new FormData();
+        console.log(this.state.oldExpertValue);
+        console.log(this.state.expertValue);
```
Comment (sudheesh001) [inline src/components/SkillEditor/SkillEditor.js]: Same here.

File: src/components/SkillEditor/SkillEditor.js [hunk 5]
```diff
@@ -148,82 +189,227 @@ export default class Container extends React.Component {
...
+          "async": true,
+          "crossDomain": true,
+          "url": "http://api.susi.ai/cms/modifySkill.json",
+          "method": "POST",
+          "processData": false,
+          "contentType": false,
+          "mimeType": "multipart/form-data",
+          "data": form
+        }
+        console.log(settings);
```
Comment (sudheesh001) [inline src/components/SkillEditor/SkillEditor.js]: Same here, you just set the `settings` object here.

File: src/components/SkillEditor/SkillEditor.js [hunk 6]
```diff
@@ -148,82 +189,227 @@ export default class Container extends React.Component {
...
+        for(var pair of form.entries()) {
+                console.log(pair[0]+ ', '+ pair[1]);
+              }
```
Comment (sudheesh001) [inline src/components/SkillEditor/SkillEditor.js]: Are you logging this for a reason? I think you can remove this for loop completely right? It's just acting as a testing procedure for you and not really doing anything else.

File: src/components/SkillEditor/SkillEditor.js [hunk 7]
```diff
@@ -148,82 +189,227 @@ export default class Container extends React.Component {
...
+                                {languages}
+                            </SelectField>
+                            <TextField
+                                floatingLabelText="Enter Skill name"
+                                floatingLabelFixed={true}
+                                style={{marginLeft:10,marginRight:10}}
+                                value={this.state.expertValue}
+                                onChange={this.handleExpertChange}
+                            />
+                            { this.state.showImage &&  <img alt="preview" id="target" style={{width:60,height:60,borderRadius:"50%",marginRight:20,border: 0}}src={this.state.image}/> }
```
Comment (sudheesh001) [inline src/components/SkillEditor/SkillEditor.js]: Can you change inline style to an external style by having a css class for this and the corresponding style over there?

**Comment by sudheesh001** [comment]
Can you please address the issues I pointed out here in a later pull request? @rishiraj824

**Comment by rishiraj824** [comment]
@sudheesh001 Sure Will make the required changes :+1:

**Pull request merged**